### PR TITLE
deprecate configbuilder & move dependencies to opt extras

### PR DESCRIPTION
**Pull request by yomach**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -18,20 +18,20 @@ readme = "README.md"
 homepage = "https://github.com/qua-platform/py-qua-tools"
 
 [tool.poetry.dependencies]
-python = ">=3.7.1,<4.0"
+python = ">=3.8,<4.0"
```
[the remark this answers is not in the document]
Comment (yomach) [reply pyproject.toml:21]: We can drop support soon

File: pyproject.toml [hunk 2]
```diff
@@ -18,20 +18,20 @@ readme = "README.md"
...
-pandas = ">=1.2.4"
 qm-qua = ">=1.1.0"
 scipy = "^1.7.1"
 dash = "^2.0.0"
-dash-html-components = "^2.0.0"
-dash-core-components = "^2.0.0"
-dash-bootstrap-components = "^1.0.0"
-dash-cytoscape = "^0.3.0"
-dash-table = "^5.0.0"
+pandas = { version = ">=1.2.4", optional = true }
```
[the remark this answers is not in the document]
Comment (yomach) [reply pyproject.toml:27]: It's actually the same in `qm-qua-sdk`:
`certifi = { version = "*", optional = true }`
```
[tool.poetry.extras]
simulation = ["certifi"]
```

File: qualang_tools/config/builder.py [hunk 3]
```diff
@@ -28,6 +29,11 @@
 class ConfigBuilder:
     def __init__(self) -> None:
         """A class used to build the configuration of an experiment with QUA"""
+        warnings.warn(
```
[the remark this answers is not in the document]
Comment (yomach) [reply qualang_tools/config/builder.py:32]: This is for CS to decide, but I think the idea is to deprecate it soon (unless some user will take ownership)

File: pyproject.toml [hunk 4]
```diff
@@ -18,20 +18,20 @@ readme = "README.md"
...
 
 [tool.poetry.dependencies]
-python = ">=3.7.1,<4.0"
+python = ">=3.8,<4.0"
 matplotlib = "^3.4.2"
 numpy = "^1.17.0"
-pandas = ">=1.2.4"
 qm-qua = ">=1.1.0"
 scipy = "^1.7.1"
 dash = "^2.0.0"
```
Comment (nulinspiratie) [inline pyproject.toml:26]: Shouldn't this also be optional?

**Comment by yomach** [comment]
@TheoLaudatQM @tomerfqm - Please approve from CS side so we'll be able to merge

**Review by TheoLaudatQM — APPROVED** [review]

**Pull request merged**